cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceRuntimeService.java

```diff
@@ -292,6 +293,17 @@ protected void startUp() throws Exception {
         Collections.sort(jarFiles);
         classpath.addAll(jarFiles);
         classpath.add("job.jar/classes");
+        // Add extra jars set in cConf
+        for (URI jarURI : CConfigurationUtil.getExtraJars(cConf)) {
+          if ("file".equals(jarURI.getScheme())) {
+            Location extraJarLocation = copyFileToLocation(new File(jarURI.getPath()), tempLocation);
+            job.addCacheFile(extraJarLocation.toURI());
+            classpath.add(extraJarLocation.getName());
```

Comment:
This can be pulled out of the `if-else` block and just do `classpath.add(LocalizationUtils.getLocalizedName(jarURI))`